src/java.base/share/classes/java/util/stream/AbstractPipeline.java

```diff
@@ -402,7 +435,7 @@ protected final boolean hasAnyStateful() {
      * operation.
      */
     @SuppressWarnings("unchecked")
-    private Spliterator<?> sourceSpliterator(int terminalFlags) {
+    protected Spliterator<?> sourceSpliterator(int terminalFlags) {
```

Comment:
Needed to make this visible to subclasses in order to implement terminal-operations.